src/tests/functional/shared_test_classes/src/subgraph/gather_weights_decompression.cpp

```diff
@@ -10,6 +10,41 @@
 namespace ov {
 namespace test {
 
+void GatherWeightsDecompressionBase::generate_inputs(const std::vector<ov::Shape>& target_input_static_shapes) {
+    inputs.clear();
+    const auto& model_inputs = function->inputs();
+    for (size_t i = 0; i < model_inputs.size(); ++i) {
+        const auto& model_input = model_inputs[i];
+        ov::test::utils::InputGenerateData in_data;
+        in_data.start_from = -1;
+        in_data.range = 2;
+        in_data.resolution = 10000;
+        ov::Tensor tensor = ov::test::utils::create_and_fill_tensor(model_input.get_element_type(),
+                                                                    target_input_static_shapes[i],
+                                                                    in_data);
+        inputs.insert({model_input.get_node_shared_ptr(), tensor});
+    }
+}
+
+void GatherWeightsDecompressionBase::check_results(const ov::element::Type& weights_precision) {
+    size_t num_exec_ops = 0;
+
+    for (const auto& n : compiledModel.get_runtime_model()->get_ordered_ops()) {
+        if (n->get_friendly_name() == "Compressed_weights") {
+            ASSERT_EQ(n->get_output_element_type(0), weights_precision);
+        }
+        if (n->get_input_size() > 0) {
+            num_exec_ops += 1;
+        }
+    }
+
+    if (weights_precision == ov::element::Type_t::f16 || weights_precision == ov::element::Type_t::bf16) {
+        EXPECT_LE(num_exec_ops, 4u);
+    } else {
+        EXPECT_LE(num_exec_ops, 3u);
+    }
```

Comment:
Let's pass expected ops count as method's parameter: as far as I understand, in `GatherWeightsDecompression` we always expect `3`, in `GatherWeightsDecompressionWithoutScale` - `4`

Reply:
done, thanks Vladislav!